PINCache/PINDiskCache.m

```diff
@@ -268,25 +270,36 @@ + (dispatch_queue_t)sharedTrashQueue
     return trashQueue;
 }
 
++ (NSLock *)sharedLock
+{
+    static NSLock *sharedLock;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sharedLock = [NSLock new];
+    });
+    return sharedLock;
+}
+
 + (NSURL *)sharedTrashURL
 {
-    static NSURL *sharedTrashURL;
-    static dispatch_once_t predicate;
+    NSURL *trashURL;
```

Comment:
nit: can you set this to nil just in case someone changes the code below in the future?